Resources/config/config.xml

```diff
@@ -58,8 +58,8 @@
             <tag name="twig.extension" alias="fos_message" />
         </service>
 
-        <service id="fos_message.search_query_factory.default" class="FOS\MessageBundle\Search\QueryFactory" scope="request" public="true">
-            <argument type="service" id="request" />
+        <service id="fos_message.search_query_factory.default" class="FOS\MessageBundle\Search\QueryFactory" public="true">
+            <argument type="service" id="request_stack" />
```

Comment:
2.3 only receives security fixes and it's really easy to upgrade to a higher version. I don't think we should give people an argument to not upgrade.